android/src/main/java/com/robinpowered/react/vitals/RNVitalsModule.java

```diff
@@ -66,13 +67,39 @@ private WritableMap getMemoryInfo() {
   }
 
   @Override
-  public void onLowMemory() {
+  public void onTrimMemory(level) {
     ReactApplicationContext context = getReactApplicationContext();
     if (context.hasActiveCatalystInstance()) {
       context.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(LOW_MEMORY, getMemoryInfo());
     }
   }
 
+  @Override
+  public void onLowMemory() {
+    // no-op
```

Comment:
What's the difference between `onTrimMemory` and `onLowMemory`? I assume one is a "pre" and one is a "post" notification? (before and after memory is collected).

If the latter, is `onTrimMemory` going to fire _after_ memory has been collected? Unlike iOS, where it fires before?

Reply:
Basically `ComponentCallbacks` with the `onLowMemory()` method was introduced in API Level 1. `onLowMemory()` would usually trigger when the device is running low on memory and requires trimming. 

Fast forward to API level 14, Android introduced `ComponentCallbacks2` with the `onTrimMemory(int level)` method that the OS calls when it has determined that it is a good time for a process to trim unneeded memory from its process, alongside with the level of severity, that hints to the amount of trimming the application may like to perform.

TDLR: Post API Level 14, it's recommended to only rely on `onTrimMemory()` but `onLowMemory()` is still required since we're implementing an interface.